documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+
+To utilize the `saveobj/loadobj` Matlab functionality we have decided to make all Horace objects `serializable`. The custom `serializable` class defines `saveobj/loadobj` pair of methods and some additional methods, necessary to maintain class versioning (see below).
+To implement our *serializable* interface we have to make two assumptions about our objects:
+
+1. The object has an empty constructor.
+2. The object has a public interface which allows to define(assign) to an empty object any contents the non-empty object may have, defining in this way any valid non-empty object.
+
+If such assumptions are satisfied, we may define *serializable* objects, which need to overload only handful of class-specific methods, but would immediately have number of very useful generic features. The class diagram describing such object is presented on the **Fig 1**:
+
+<center>![Serialiable. Main Interface](../diagrams/serializable_main_interface.png)</center>
+
+**Fig 1:** Main interface and methods of **serializable** class.
+
+The parent class, presented on the **Fig 1**, allows easy construction and maintenance of standardized serializable objects by overloading the following abstract methods:
+
+**Table 1: Main abstact methods requesing overloading (definition)**
+
+| Method | Description |
+|-----|---------|
+|  classVersion | returns the number, which describes current class version. E.g. 1 - for version 1|
+| saveableFields | returns list of property names, defining the class public interface |
+
+As soon as these properties are defined, one can use other class methods to serialize/deserialize objects and convert objects to/from structure. The main features of the available methods are summarized in the Tables 2:
+** Table 2 Main methods, available for serializable objects:**
+
+| Method | Description |
+|-----|---------|
+|  `struct = to_bare_struct(obj)` | returns structure, with the field names equal to the names returned by `saveableFields` method and values, defined by the values of these fields. If the values of fields are the serializable objects themselves, they are converted into correspondent structures recursively |
+| `obj = from_bare_struct(obj,data)` | Reverse of `to_bare_struct` method. Sets the values of the public properties of the serializable interface from the structure, obtained from `to_bare_struct` method |
+| `data = to_struct(obj)` | converts `obj` or array of `obj` into the `data` structure using `to_bare_struct` method and adds complimentary information about the class name and the size and shape of the `obj` array if relevant. |
```

Comment:
complimentary->complementary (complImentary would be "wow, this is a really cute struct")
I would have said additional, but your call on that